### ISSUE TITLE
The null character is included in the HTTP response due the incorrect buffer length

### ISSUE THREAD

**Issue by loadhigh** [dlundquist/sniproxy#298]
https://github.com/dlundquist/sniproxy/blob/7fa03db7d28957493326f8a7af8e23487e25c6e5/src/http.c#L51

Consider using strlen() instead of sizeof

**Comment by dlundquist** [comment]
Thanks for pointing this out. I don’t think strlen will work in this initialization context but sizeof - 1 will.

**Comment by loadhigh** [comment]
I'm sure strlen() call will be optimized by gcc into a literal just like sizeof at least for gcc

**Comment by dlundquist** [comment]
Clang forbids it:
```
[dustin@dawn sniproxy (master)]$ make
Making all in src
make[1]: Entering directory '/home/dustin/src/sniproxy/src'
  CC       http.o
http.c:51:26: error: initializer element is not a compile-time constant
    .abort_message_len = strlen(http_503),
                         ^~~~~~~~~~~~~~~~
1 error generated.
Makefile:424: recipe for target 'http.o' failed
```

Going with `sizeof(http_503) - 1`, thanks for bringing this to my attention.

**Comment by dlundquist** [comment]
Fixed in master f35b6a7.
